core/container/pool/pool.odin

```diff
@@ -0,0 +1,96 @@
+package container_pool
+
+import "core:fmt"
+
+// Single bit to mark a slot as active
+PoolFlag :: bit_set[0..<64;u64]
+
+// Dynamically resizable generic data pool
+Pool :: struct($type: typeid) {
+    data : [dynamic]type,
+    flag : [dynamic]PoolFlag,
+    free : [dynamic]int
+}
+
+// Reserve memory for the pool
+init :: proc($type: typeid, data_cnt : uint, free_cnt : uint = 8) -> Pool(type) {
```

Comment:
Core library uses `int` type for all indices and sizes. This makes computations involve less casts.